codegenerator/cli/templates/static/codegen/src/ReorgDetection.res

```diff
@@ -178,6 +178,19 @@ module LastBlockScannedHashes: {
     ~confirmedBlockThreshold: int,
   ) => lastBlockScannedData.blockNumber < currentHeight - confirmedBlockThreshold
 
+  type rec trampoline<'a> = Data('a) | Callback(unit => trampoline<'a>)
+
+  /**
+    Trampolines are a method of handling mutual recursions without the risk of hitting stack limits
+
+    Tail Call Optimization is not possible on mutually recursive functions and so this is a manual optizimation
+  */
+  let rec trampoline = value =>
```

Comment:
@DZakh does the comment help?

Reply:
Could you add that it works by generating JS with a `while` loop